tasks/data_handling/adapted_gene_embeddings.py

```diff
@@ -0,0 +1,95 @@
+import scanpy as sc
+import numpy as np
+import os
+from typing import Tuple, List
+
+
+def find_common_genes(
+        folder_path: str = None,
+        save_common_genes: bool = False,
+        save_path: str = 'intersection_genes'
+) -> List:
+    """
+    ### Function that returns a list of genes available in every h5-files found in folder_path
+
+    ---
+
+    #### Args:
+        - folder_path (str): Path to folder with all .h5-files. Default None
+        - save_common_genes (bool): If True, saves list of genes available in all .h5-files under save_path.
+            Default False
+        - save_path (str): Where to save the list, not necessary if save_common_genes=False.
+            Default 'intersection_genes'
+
+    ---
+    #### Returns:
+        - List of genes that are in all .h5-files from folder_path.
+    ---
+    Written: ronjah@chalmers.se
+    """
+    files = os.listdir(folder_path)
+    file_names = [file for file in files if file.endswith('.h5')]
+    adata = sc.read_h5ad(folder_path + '/' + file_names[0])
+    common_genes = adata.var_names.to_list()
+    # Checks intersection of genes between .h5-files in folder_path
+    for file in file_names:
+        adata = sc.read_h5ad(folder_path + '/' + file)
+        gene_list_file = adata.var_names.to_list()
+        common_genes_set = set(common_genes)
+        common_genes = set(gene_list_file) & common_genes_set
+    if save_common_genes:
+        np.save(save_path, common_genes)
```

Comment:
This should have an extension. I'm guessing .npy?

Reply:
It adds that by itself